src/providers/WorkflowCore.Persistence.EntityFramework/Services/EntityFrameworkPersistenceProvider.cs

```diff
@@ -185,9 +187,10 @@ public async Task PersistWorkflow(WorkflowInstance workflow, List<EventSubscript
 
                 foreach (var subscription in subscriptions)
                 {
-                    subscription.Id = Guid.NewGuid().ToString();
+                    subscription.Id = Guid.Empty.ToString();
                     var subscriptionPersistable = subscription.ToPersistable();
-                    db.Set<PersistedSubscription>().Add(subscriptionPersistable);
+                    var result = db.Set<PersistedSubscription>().Add(subscriptionPersistable);
```

Comment:
also, there no need for re-assign as it not used anywhere
u can just leave as it was